docs/data/data-grid/export/CustomExport.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+import { useDemoData } from '@mui/x-data-grid-generator';
+import {
+  DataGrid,
+  GridToolbarContainer,
+  GridToolbarContainerProps,
+  GridToolbarExportContainer,
+  GridCsvExportMenuItem,
+  GridCsvExportOptions,
+  GridExportMenuItemProps,
+  useGridApiContext,
+  gridFilteredSortedRowIdsSelector,
+  visibleGridColumnsSelector,
+} from '@mui/x-data-grid';
+import { CssBaseline } from '@mui/material';
+import MenuItem from '@mui/material/MenuItem';
+import { ButtonProps } from '@mui/material/Button';
+
+const getJson = (apiRef) => {
+  // Select rows and columns
+  const filteredSortedRowIds = gridFilteredSortedRowIdsSelector(apiRef);
+  const visibleColumnsField = visibleGridColumnsSelector(apiRef).map(
+    (col) => col.field,
+  );
+
+  // Format the data. Here we only keep the value
+  const data = filteredSortedRowIds.map((id) => {
+    const row = {};
+    visibleColumnsField.forEach((field) => {
+      row[field] = apiRef.current.getCellParams(id, field).value;
+    });
+    return row;
+  });
+
+  // Stringify with some indentation
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#parameters
+  return JSON.stringify(data, null, 2);
+};
+
+const exportBlob = (blob, filename) => {
+  // Save the blob in a json file
+  const url = URL.createObjectURL(blob);
+
+  const a = document.createElement('a');
+  a.href = url;
+  a.download = filename;
+  a.click();
+
+  setTimeout(() => {
+    URL.revokeObjectURL(url);
+  });
+};
+
+export const JsonExportMenuItem = (props: GridExportMenuItemProps<{}>) => {
+  const apiRef = useGridApiContext();
+
+  const { hideMenu } = props;
+
+  return (
+    <MenuItem
+      onClick={() => {
+        const jsonString = getJson(apiRef);
+        const blob = new Blob([jsonString], {
+          type: 'text/json',
+        });
+        exportBlob(blob, 'DataGrid_demo.json');
+
+        // Hide the export menu after the export
+        hideMenu?.();
+      }}
+    >
+      Export JSON
+    </MenuItem>
+  );
+};
+
+const csvOptions: GridCsvExportOptions = { delimiter: ';' };
+
+const CustomExportButton = (props: ButtonProps) => (
+  <GridToolbarExportContainer {...props}>
+    <GridCsvExportMenuItem options={csvOptions} />
+    <JsonExportMenuItem />
+  </GridToolbarExportContainer>
+);
+
+const CustomToolbar = (props: GridToolbarContainerProps) => (
```

Comment:
```suggestion
const CustomToolbar = (props: GridToolbarProps) => (
```

Should we use the correct interface here to give a hint to the user that they can still use the `csvOptions` and `printOptions` prop here ?

Reply:
I don't think so, because it implies implementing the all pipeline.
They have to pass the `csvOptions` prop of `CustomToolbar` to `CustomExportButton` which pass it to `GridCsvExportMenuItem`

So I prefer to only show the type of props used. I hoe that without explicitly showing the exportOptions, it will free their imagination, to add prop such as `exportedFields` which ended with `printOptions={fields: exportedFields}` and `csvOptions={fields: exportedFields}`